src/rgw/rgw_user.cc

```diff
@@ -1391,13 +1391,18 @@ int RGWSubUserPool::add(RGWUserAdminOpState& op_state, std::string *err_msg, boo
 {
   std::string subprocess_msg;
   int ret;
+  int32_t key_type = op_state.get_key_type();
 
   ret = check_op(op_state, &subprocess_msg);
   if (ret < 0) {
     set_err_msg(err_msg, "unable to parse request, " + subprocess_msg);
     return ret;
   }
 
+  if (key_type == KEY_TYPE_S3 && op_state.get_access_key().empty()) {
```

Comment:
Maybe We don't need to check KEY_TYPE here? Found the method "RGWAccessKeyPool::generate_key(...)" will add access key for SWIFT type user automatically